### PR TITLE
163383129 Setup react router v4

### PR DESCRIPTION
**Pull request by codjoero**
#### Description
---
PR to deliver a task to Setup react-router v4 in the frontend project

#### Tasks completed
---
- install react-router-dom
- add views folder
- add components folder
- add routes folder
- add home and login routes

#### How can this be tested?
---
Clone the repository
Run git checkout *ch-setup-react-router-163383129* to change to this branch
Install dependencies by running npm or yarn
Start the Application with yarn start or npm start
On the client, navigate between home and login

#### What are the relevant Pivotal Tracker stories?
---
[#163383129](https://www.pivotaltracker.com/story/show/163383129)

### CODE REVIEW

File: src/components/App.jsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import React, { Component } from 'react';
+import '../styles/App.css';
+
+class App extends Component {
```
Comment (karuhanga) [inline src/components/App.jsx:4]: I think we'll need to agree on a convention for presentation components.
This PR is currently `mixing` `class` based `and` `functional` components.

**Review by karuhanga — CHANGES_REQUESTED** [review]
Thumbs up on the pace with which this got done.

A few observations though;
- I would need for you to clarify on the name `views`. I don't feel it is a good name for what those components `will be doing`.
- Those components should ideally be `class based`, given they will use `life cycle` methods for things like the initial loading of data.

File: src/components/App.jsx (hunk 1 above)
Comment (codjoero) [reply src/components/App.jsx:4]: @Karuhanga thanks for your review, may I refer you to the [story rubric](https://www.pivotaltracker.com/story/show/163383129)  for the structure, and I believe we will definitely use both class and functional components

Comment (missvicki) [reply src/components/App.jsx:4]: @Karuhanga from my understanding, one uses the functional components when not interacting with the state and class components when interacting with the state.

**Review by missvicki — APPROVED** [review]
LGTM

File: src/routes/index.jsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
+import Header from '../components/Header';
+import HomeView from '../views/HomeView';
+import LoginView from '../views/LoginView';
+
+const Routes = () => (
+    <Router>
+        <div>
```
Comment (fahdjamy) [inline src/routes/index.jsx:9]: Great work you've done here @codjoero.
Just on this Router functional component, I wanted to as is the Login page also going to have a Header (Navbar) or should this be this be conditional and appear on pages like home, profile among others.
I mean based off of what Vicky did in the mockups.

File: src/components/Header.jsx [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
```
Comment (fahdjamy) [inline src/components/Header.jsx]: Good work @codjoero.
I see the Header component is indirectly in the components folder.
Don't you think it would be better off if we categorize components in terms of what they are representing?
In this case, I'd rather the Header component to be under a sub-folder. e.g **src/components/UI/Header.jsx** because a header component is more of a UI component.

**Review by fahdjamy — CHANGES_REQUESTED** [review]
Thanks for the good work @codjoero. 
I have left a few comments.
Plus I have read the description for the task where basically the view is like our "container" type of folder so don't you think everything in the views folder would be better off as a class-based component and not functional ones?

File: src/components/App.jsx (hunk 1 above)
Comment (karuhanga) [reply src/components/App.jsx:4]: Not necessarily. Hooks allow us access state from functions as well.

That said though, I have talked to @codjoero and @FahdJamy, I'd like everyone else's opinion as well, but the consensus is in the future;
- We'll stick to stateless functional components for presenters
- If a component needs state, we'll wrap it in a container to manage state for it

I hope the rationale for this makes sense.

**Review by karuhanga — APPROVED** [review]
With the discussions we've had, Looks Good.

File: src/components/App.jsx (hunk 1 above)
Comment (oma0256) [reply src/components/App.jsx:4]: @Karuhanga yes I think we should stick with that convention

**Comment by oma0256** [comment]
Good work @codjoero But I think I agree with @FahdJamy that the components in views should be class based and not functional based otherwise good work

**Review by fahdjamy — APPROVED** [review]

**Review by oma0256 — APPROVED** [review]

**Comment by pbkabali** [comment]
@codjoero, thanks for the explanation. I was used to the convention where the main `App.jsx` file is the one rendered into the DOM.

**Comment by collinewait** [comment]
Resolve conflicts @codjoero

File: .eslintrc.json [hunk 4]
```diff
@@ -37,7 +37,7 @@
     "react/jsx-boolean-value": 1,
     "react/jsx-closing-bracket-location": 0,
     "react/jsx-curly-spacing": 1,
-    "react/jsx-indent-props": 0,
+    "react/jsx-indent-props": ["error", 4],
```
Comment (collinewait) [inline .eslintrc.json]: `eslint` changes are not related to the task.

Comment (collinewait) [reply .eslintrc.json]: You can revert them

**Review by collinewait — APPROVED** [review]

**Review by collinewait — APPROVED** [review]
